mmdet/models/utils/res_layer.py

```diff
@@ -56,22 +59,43 @@ def __init__(self,
             downsample = nn.Sequential(*downsample)
 
         layers = []
-        layers.append(
-            block(
-                inplanes=inplanes,
-                planes=planes,
-                stride=stride,
-                downsample=downsample,
-                conv_cfg=conv_cfg,
-                norm_cfg=norm_cfg,
-                **kwargs))
-        inplanes = planes * block.expansion
-        for i in range(1, num_blocks):
+        if downsample_first:
             layers.append(
                 block(
                     inplanes=inplanes,
                     planes=planes,
-                    stride=1,
+                    stride=stride,
+                    downsample=downsample,
+                    conv_cfg=conv_cfg,
+                    norm_cfg=norm_cfg,
+                    **kwargs))
+            inplanes = planes * block.expansion
+            for _ in range(num_blocks - 1):
```

Comment:
`for _ in range(1, num_blocks)`

Reply:
The two cases are different. Use `range(1, num_blocks)` for the last `num_blocks - 1` blocks and `range(num_blocks - 1)` for the first `num_blocks - 1` blocks.